packages/node/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -50,8 +50,17 @@ import { KeyObject } from "crypto";
 import { fetch as globalFetch } from "cross-fetch";
 
 import { EventEmitter } from "events";
+import { Headers as NodeHeaders } from "node-fetch";
 import { configToIssuerMetadata } from "../IssuerConfigFetcher";
 
+// Using the cross-fetch polyfill should enforce Headers to be present in the Node environment
+// However, it does not behave as expected, which is why the global.Headers value
+// is manually overwritten.[]
+// import "cross-fetch/polyfill";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+global.Headers = NodeHeaders;
+
```

Comment:
This was bugging me a bit.
I think we probably don't need that.
What might be enough is to import both Headers and fetch in the core package fetchFactory: packages/core/src/authenticatedFetch/fetchFactory.ts
That is: `import { Headers, fetch } from "cross-fetch";` (instead of `import { fetch }...`).

The TypeScript compiler gets confused by the first line in "cross-fetch" exports `/// <reference lib="dom" />`:
https://github.com/lquixada/cross-fetch/blob/dcb9279a50f911c4a6e031de44397305a7c948a6/index.d.ts#L1-L14

Which I think makes it believe the whole lib.dom is available ^^,